pallets/dca/src/lib.rs

```diff
@@ -413,7 +417,7 @@ pub mod pallet {
 		/// Emits `Scheduled` and `ExecutionPlanned` event when successful.
 		///
 		#[pallet::call_index(0)]
-		#[pallet::weight(<T as Config>::WeightInfo::schedule())]
+		#[pallet::weight(<T as Config>::WeightInfo::schedule() + <T as Config>::AmmTradeWeights::calculate_buy_trade_amounts_weight(schedule.order.get_route()))]
```

Comment:
strange to see here buy specific info, namely `calculate_buy_trade_amounts_weight`

What if we schedule a DCA with sell?

Reply:
Hmm. We don't call the router for sell. Only for buy. This can be optimized to add the weight only if trade == buy.

Reply:
I wonder how much we gain with that optimization. If you think we would gain a lot of saved ref time, we could optimize it, otherwise just leave.

Reply:
i dont understand why is even here, the route shouldn't change weight of the schedule extrinsic as the execution is not even done here

Reply:
Because the extrinsic calls `Self::get_amount_in_for_buy()` which calls `Router::calculate_buy_trade_amounts()`